Examples/Run/Fatras/Common/FatrasDigitizationBase.cpp

```diff
@@ -26,40 +29,71 @@ void ActsExamples::setupDigitization(
   // Read the standard options
   auto logLevel = ActsExamples::Options::readLogLevel(vars);
 
-  // Configure the digitizer
-  ActsExamples::DigitizationAlgorithm::Config digi;
-  digi.inputSimulatedHits = "hits";
-  digi.outputClusters = "clusters";
-  digi.planarModuleStepper = std::make_shared<Acts::PlanarModuleStepper>(
-      Acts::getDefaultLogger("PlanarModuleStepper", logLevel));
-  digi.randomNumbers = randomNumbers;
-  digi.trackingGeometry = trackingGeometry;
-  sequencer.addAlgorithm(
-      std::make_shared<ActsExamples::DigitizationAlgorithm>(digi, logLevel));
+  if (vars["digi-smearing"].as<bool>()) {
+    ActsExamples::SmearingAlgorithm::Config smearCfg =
+        ActsExamples::Options::readSmearingConfig(vars);
+    smearCfg.trackingGeometry = trackingGeometry;
+    smearCfg.randomNumbers = randomNumbers;
 
-  // Output directory
-  std::string outputDir = vars["output-dir"].template as<std::string>();
+    sequencer.addAlgorithm(
+        std::make_shared<ActsExamples::SmearingAlgorithm>(smearCfg, logLevel));
 
```

Comment:
I'm thinking how we could use the `outputMeasurements` for the reco. Do we just include the `SmearingAlgorithm` in the reco or we need to have a Csv writer/reader for the `outputMeasurements` so that they could be used directly in the reco?

Reply:
Let me first bring the planar/geometric digitisation onto the same footing, then we sort out the example flow.

Reply:
My proposal would be to remove the digitization from the Fatras simulation and move it to the reconstruction examples. That way we need to simulate only once, but run different scenarios for reconstruction. If we see that this takes too much time as well, we can add a separate digitization step that writes out the measurements and a reader for the reconstruction examples.

Reply:
Sounds good, as I said, I would like to do this after the geometric digitisation is in.